src/app/pages/wallets/my-wallets/pages-wallets-my-wallets.component.ts

```diff
@@ -26,15 +30,65 @@ export class PagesWalletsMyWalletsComponent implements OnInit {
           data,
           isLoading: false,
           hasError: false,
+          nowEditWallet: 0,
         };
       },
       error: () => {
         this.myWalletsData = {
           data: null,
           isLoading: false,
           hasError: true,
+          nowEditWallet: 0,
         };
       },
     });
   }
+
+  public addNewWallet(userInputData_: string) {
+    this.myWalletsService.addNewWallet(userInputData_).subscribe( (wallet_: MyWallet) => {
+        this.myWalletsData.data?.push(wallet_);
+    });
+  }
+
+  public updateWallet(userInputData_: string) {
+    this.myWalletsService.updateNewWallet(this.myWalletsData.nowEditWallet, userInputData_).subscribe( (wallet_: MyWallet) => {
+        const walletIndex = this.myWalletsData.data?.findIndex(w => w.id === this.myWalletsData.nowEditWallet);
```

Comment:
It can be simpler:
```typescript
this.myWalletsData.data = this.myWalletsData.data.map(wallet => {
    if (wallet.id === editedWallet.id) {
        return editedWallet;
    }
  
    return wallet;
})
```